### ISSUE TITLE
pkg_jem_v4.0.1: what is minimum booked/user=0? and other calculations

### ISSUE THREAD

**Issue by jojo12** [jemproject/JEM-Project#1630]
I am testing:
1) I think  min/user=0 is wrong: only if you are a ghost, you can book attending with 0 places!

2) In addition: if max places=0 (unlimited) then reserved places =0. This is wrong too
Even if there is no limit of places, I would like to have the possibility to have reserved places, because
 I had always persons, who didn't want to register in joomla, but want to attend and use other possibilities like letter, phone.
As organisator I can use reserved places as reminder.

The request answer branch is not yet committed: I need a combined zip to test. (there are changes on the same files in different branches.

**Comment by mckillo** [comment]
Where do you see min/user=0?

**Comment by mckillo** [comment]
I see 10 conflicts to the merge, I'll resolve it and merge 'request-answer with online-help' into 'JEM-4.0.1-dev'. Let's some minutes.

**Comment by mckillo** [comment]
Done, you can create a 4.0.1 zip with all branchs (ready to test it and public)

**Comment by jojo12** [comment]
done and installed:
1) Problem:
I am in backend>event>registration
I want max.places=0 (unlimited)
then I want min.places/user=1 , but the standard is 0 (zero), I choose 1, then it changes back to 0 AND max.place/user changes to 1. I can't change back 

2) problem: waitinglist check is now doubled!

**Comment by jojo12** [comment]
the legacy frontend>event page is broken (all on the left!)

**Comment by jojo12** [comment]
it must be inside of line 163-201

**Comment by mckillo** [comment]
Solved.
Fix error in this view by last merge of branchs. Now, the conflict is right..

**Comment by jojo12** [comment]
Yes, I tested: Problem 3) is solved.
Persists:
1) There are now in backend and frontend eventedit: when max.places=unlimited, then user can be a ghost (places =0)
2) Min.places/user can never be =0

**Comment by mckillo** [comment]
There is only one wailing list option in the registration section.
Improvement style of Available Places (label and input in the same line).
See commit above (one less)

**Comment by mckillo** [comment]
Respect to the registration, the javascript is wrong. If I edit an event and write down Max Places = 0, all fields reset to 0.

**Comment by mckillo** [comment]
@hekla Could you review the code of javascript? See previous image and issue about Max Places (reset the other fields, when write 0)

**Comment by jojo12** [comment]
@mckillo : problem doubled waitinglist is solved, thanks!

**Comment by jojo12** [comment]
next problem: backend>event>registration: free places are ---, when max.places=0 (unlimited)

**Comment by jojo12** [comment]
Sorry, but this whole thing is sh...! You wanted to do it better, but it is much worse!

**Comment by jojo12** [comment]
can we go back?

**Comment by jojo12** [comment]
@Heklaterriol : (falls Du das überhaupt liest:
bisher war es so:
maximal Plätze = eine Zahl für die Anzahl limitierter Plätze oder wenn Null für Unbegrenzt
reserved places = vorenthaltene Plätze, z.B. für unvorhergesehenes (gibts immer, oder anderweitig angemeldete.
minimum pro user, kann nicht Null sein (sonst Geist) sollte standard sein.
wenn Antwort (Absagen) erwartet werden 
Warteliste, entweder wenn voll oder von Organisator draufgesetzt.
Du kannst also nicht alles durchrechnen.

**Comment by mckillo** [comment]
I have uploaded a commit (see above): Fix the available field as a number (in input and field xml form), with a default value of 0 when placesmax=0 (add the meaning to the help tag too).

@hekla I have added some modifications to your Javascript code:

Lines 201-204 eliminate the same condition as lines 198–200 (same condition and action).
Lines 192, 195, and 104 add other conditionals && maxPlacesvalue !=0. I told you above that when I write '0' in Placesmax, it resets other fields.

Review the JavaScript code because there are errors before line 171. I saw in Github that you changed during beta1. See the console in the browser because there are errors in javascript, and the last check of the registration field doesn't work for those errors.
